README.md

```diff
@@ -1,2 +1,154 @@
+
+```shell
+nix flake update
+```
+
+This will update all inputs to the latest version. In case you just want to update an individual input, append the input identifier to the above command.
+
+```shell
+nix flake update holonix
+```
+
+The next time you enter the dev shell, the updated binaries will be downloaded and provided in the resulting shell. Should you have run `nix flake update` from inside the dev shell, you need to `exit` the shell first and then enter it again with `nix develop`.
+
+[nix flake update in Nix Reference Manual](https://nix.dev/manual/nix/latest/command-ref/new-cli/nix3-flake-update)
+
+## Including other versions of Holochain
+
+By default the Holonix input references the `main` branch of the Holonix repository. Whatever the current revision of that branch at the moment of invoking the dev shell for the first time is will be written to the `flake.lock` file. The `main` branch uses version 0.4 of Holochain and compatible versions of scaffolding etc.
+
+To use other versions of Holochain in the dev shell, the `holonix` input must be modified.
+
+```nix
+inputs = {
+    holonix.url = "github:holochain/holonix?ref=main";
+    ...
+};
+```
+
+For Holochain v0.3, change it to `main-0.3`.
+
+```nix
+inputs = {
+    holonix.url = "github:holochain/holonix?ref=main-0.3";
+    ...
+};
+```
+
+Now running `nix develop` will update the flake lock file with the current revision of the `main-0.3` branch of the Holonix repository and enter the dev shell.
+
+## Defining packages to include in the dev shell
+
+It may be that you want to add or remove packages included in the dev shell. This is the relevant section
+
+```nix
+...
+packages = (with inputs'.holonix.packages; [
+    holochain
+    lair-keystore
+    hc-launch
+    hc-scaffold
+    hn-introspect
+    rust # For Rust development, with the WASM target included for zome builds
+])
+...
+```
+
+If you wanted to only use the Holochain binaries in a dev shell, all other packages can be removed from the default shell.
+
+```nix
+...
+packages = (with inputs'.holonix.packages; [
+    holochain
+    rust # For Rust development, with the WASM target included for zome builds
+])
+...
+```
+
+Other packages from the Nix packages collection can be added here.
+
+```nix
+]) ++ (with pkgs; [
+    nodejs_20 # For UI development
+    binaryen # For WASM optimisation
+    # Add any other packages you need here
+]);
+```
+
+For example the `jq` package can be added.
+
+```nix
+]) ++ (with pkgs; [
+    nodejs_20 # For UI development
+    binaryen # For WASM optimisation
+    # Add any other packages you need here
+    jq
+]);
+```
+
+> Note that by default Rust is included in the dev shell. If you want to use your globally installed version of Rust instead, remove the `rust` package from the list of packages.
```

Comment:
Useful note but please also mention that using a newer version of Rust than Holochain is currently using might result in warnings or build issues. People are welcome to report those on `holochain/holochain`